### PR TITLE
Muffler refactor

### PR DESCRIPTION
**Pull request by leagris**
Followup of https://github.com/GTNewHorizons/GT5-Unofficial/pull/459

Refactor + translatable description

### CODE REVIEW

**Comment by leagris** [comment]
Tested ok:

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java [hunk 1]
```diff
@@ -6,17 +6,29 @@
...
+            "gt.blockmachines.hatch.muffler.desc.format",
+            "Outputs the Pollution (Might cause ... things)%n" +
+                    "DO NOT OBSTRUCT THE OUTPUT!%n" +
+                    "Reduces Pollution to %d%%%n" +
+                    "Recovers %d%% of CO2/CO/SO2");
+    private final int pollutionReduction = calculatePollutionReduction(100);
+    private final int pollutionRecover = 100 - pollutionReduction;
+    private final String[] description = String.format(localizedDescFormat, pollutionReduction, pollutionRecover)
+            .split("\\R");
+    private final boolean[] facings = new boolean[]{false, true, true, true, true, true};
```
Comment (basdxz) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java:30]: Unsure if this breaks with what @bartimaeusnek  and @Technus were doing, shouldn't I think.

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java [hunk 2]
```diff
@@ -173,4 +143,43 @@ public void pollutionParticles(World aWorld, String name) {
...
+        if (getBaseMetaTileEntity().getAirAtSide(getBaseMetaTileEntity().getFrontFacing())) {
+            GT_Pollution.addPollution(getBaseMetaTileEntity(), calculatePollutionReduction(10000));
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param aFacings the {@link ForgeDirection} invalid facings
+     * @apiNote API Code, BartWorks/TecTech based EBF relies on this. It's marked here, not anywhere else.
```
Comment (basdxz) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java:178]: Did not know we had an API note thing

**Review by basdxz — APPROVED** [review]

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java (hunk 1 above)
Comment (bartimaeusnek) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java:30]: id prefer to not have it this abstract. initial value should be set in init with the approriate method instead. this make it less understandable whats the inital state is.

Comment (bartimaeusnek) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Muffler.java:30]: id prefer to not have it this abstract. initial value should be set in init with the approriate method instead. this make it less understandable whats the inital state is.

**Review by bartimaeusnek — CHANGES_REQUESTED** [review]

**Pull request merged**